sim/core/aura.go

```diff
@@ -329,124 +324,140 @@ func (at *auraTracker) AddAura(sim *Simulation, newAura Aura) {
 		panic("Empty aura ID")
 	}
 
-	if aura := at.auras[newAura.ID]; aura.ID != 0 {
+	if newAura.Duration == 0 {
+		panic("Aura with 0 duration")
+	}
+
+	if aura := &at.auras[newAura.ID]; aura.ID != 0 {
 		// Getting lots of bug reports, do a grep and catch all cases for this before uncommenting.
 		//panic(fmt.Sprintf("AddAura(%v) at %s - previous has %s left, use ReplaceAura() instead", newAura.ActionID, sim.CurrentTime, aura.expires-sim.CurrentTime))
 		at.RemoveAura(sim, newAura.ID)
 	}
 
-	newAura.startTime = sim.CurrentTime
-	newAura.Refresh(sim)
+	if newAura.Duration == NeverExpires {
+		newAura.expires = NeverExpires
+		newAura.startTime = sim.CurrentTime
 
-	at.auras[newAura.ID] = newAura
-	at.auras[newAura.ID].activeIndex = int32(len(at.activeAuraIDs))
-	at.activeAuraIDs = append(at.activeAuraIDs, newAura.ID)
+		// permanent auras can be ignored for advance()
+	} else {
+		newAura.startTime = sim.CurrentTime
+		newAura.expires = sim.CurrentTime + newAura.Duration
+
+		newAura.activeIndex = int32(len(at.activeAuraIDs))
+		at.activeAuraIDs = append(at.activeAuraIDs, newAura.ID)
+	}
```

Comment:
Looks like you can assign startTime and expires outside of this 'if'.

Reply:
Could do for startTime, I thought this is more stupid aka more readable. I accidentally mixed up the order, though.

Reply:
Adressed: changed so it works for similarly to RemoveAura(), and just handles the "Duration != NeverExpires" case. "expires" is never used for such auras, so it remains unset now.